app/src/main/java/org/mozilla/fenix/components/TabCollectionStorage.kt

```diff
@@ -80,6 +82,7 @@ class TabCollectionStorage(
 
     fun removeCollection(tabCollection: TabCollection) {
         collectionStorage.removeCollection(tabCollection)
+        Collections.totalCollections.set(getTabCollectionsCount().toLong())
```

Comment:
`getTabCollectionsCount()` seems to touch the DB (but I don't know how Room works) and I don't like that performance hit if so – is Room actually pulling it from in memory or is there an in-memory representation we can use instead?

Reply:
Seems like its querying the DB each time. I will look for an alternate

Reply:
Added  an observer to report only once the data is in memory